src/testing.py

```diff
@@ -89,6 +89,46 @@ def __call__(self, test_file):
       return Result(test=basename, success=False, output=e.output)
 
 
+def parse_exclude_files(fails, config_attributes):
+  ''' Parse the files containing tests to exclude (i.e. expected fails).
+  Each line may contain a comma-separated list of attributes restricting
+  the test configurations which are expected to fail. (e.g. JS engine
+  or optimization level). A test is only excluded if the configuration
+  has all the attributes specified in the exclude line. Lines which
+  have no attributes will match everything, and lines which specify only
+  one attribute (e.g. engine) will match all configurations with that
+  attribute (e.g. both opt levels with that engine). Returns a sorted list
+  of exclusions which match the attributes.
+  '''
+  excludes = {}  # maps name of excluded test to file from whence it came
+  if not config_attributes:
```

Comment:
Would a default argument make sense instead?

Microoptimization: I feel like this should be checking `is None`, because if you call this with an empty set or array, we'll allocate a new empty set, which is work we don't need to do.